crates/harness-tests/src/routing.rs

```diff
@@ -199,3 +201,36 @@ async fn monolith_double_load_room(ctx: &mut TestRunner) {
     let t = resp.text().await.expect("failed to read http response");
     assert_eq!(t, "{\"name\":\"foo\"}");
 }
+
+#[test_context(TestRunner)]
+#[tokio::test]
+async fn unicast_messaging(ctx: &mut TestRunner) {
+    let mut m = MonolithBuilder::new()
+        .behavior(BehaviorTrackClients)
+        .build(ctx)
+        .await;
+
+    m.show().await;
+
+    let mut c1 = Client::new(ctx).unwrap();
+    c1.join("foo").await;
+    let mut c2 = Client::new(ctx).unwrap();
+    c2.join("foo").await;
+
+    m.wait_recv().await;
+
+    let c_id = m.clients().iter().next().copied();
+
+    m.send(M2BRoomMsg {
+        room: "foo".into(),
+        client_id: c_id,
+        payload: RawValue::from_string("{}".to_owned()).unwrap(),
+    })
+    .await;
+
+    let res1 = c1.recv().await.unwrap();
+    let res2 = c2.recv().await.unwrap();
```

Comment:
`recv()` will time out if no message is received and return `Err` instead of `Ok`.

Instead of unwrapping them, you could put them in a Vec and make assertions on that. Something like this:
```rust
    let res = vec![c1.recv().await, c2.recv().await];
```
Then you just check to see if only one of the results in the list is `Ok` and matches the message.